src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -279,6 +286,37 @@ public function addClassResource(\ReflectionClass $class)
     }
 
     /**
+     * Checks whether the requested class, interface or trait exists and registers the result for resource tracking.
+     *
+     * @param string $class
+     *
+     * @return bool
+     *
+     * @final
```

Comment:
Then imo we should have a solution that wouldn't throw a deprecation when used as a proxy. What about checking for a keyword such as `mockable` and don't throw a deprecation when common proxy interfaces are implemented?